dataframe.py

```diff
@@ -0,0 +1,270 @@
+# MIT License
+#
+# Copyright (c) 2020 Wes McKinney
+
+from abc import ABC, abstractmethod
+from collections.abc import Mapping, MutableMapping
+from typing import Any, Hashable, Iterable, Sequence
+
+# ----------------------------------------------------------------------
+# A simple data type class hierarchy for illustration
+
+
+class DataType(ABC):
+    """
+    A metadata object representing the logical value type of a cell in a data
+    frame column. This metadata does not guarantee an specific underlying data
+    representation
+    """
+    def __eq__(self, other: 'DataType'):
```

Comment:
getting a mypy error here...
```
pandas\wesm\dataframe.py:19: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the 
argument type as "object"
pandas\wesm\dataframe.py:19: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
pandas\wesm\dataframe.py:19: note:     def __eq__(self, other: object) -> bool:
pandas\wesm\dataframe.py:19: note:         if not isinstance(other, DataType):
pandas\wesm\dataframe.py:19: note:             return NotImplemented
pandas\wesm\dataframe.py:19: note:         return <logic to compare two DataType instances>
```

should we follow the mypy recommedation?

Reply:
> should we follow the mypy recommedation?

I see no reason not to